### DISCUSSION TITLE
Accept session in addition to request

### DISCUSSION THREAD

**Discussion by ngbrown** [sergiodxa/remix-auth#142] [Ideas]
`isAuthenticated` and `logout` take a `Request` just to get the session. The page very well could have already gotten a session and if it is a storage based session, then it would be advantageous to not fetch from the database multiple times.

My suggestion is to change the signature like this:

```ts
  async logout(
    request: Request | Session,
    options: { redirectTo: string }
  ): Promise<void> {
    const session = isSession(request)
      ? request
      : await this._sessionStorage.getSession(request.headers.get("Cookie"));
// the rest is identical ...
}
```

`isSession` is provided by `remix`.

**Comment by sergiodxa** [comment]
This is a good idea, I think I could add this feature.
